File: springboot-netty-rpc2/rpc-server/src/test/java/pres/test/Test1.java
package pres.test;

import org.junit.Test;

/**
 * @author HJC
 * @version 1.0
 * 谦谦君子 卑以自牧也
 * @date 2020/7/22  17:44
 * @description :
 */
public class Test1 {

    @Test
    public void m1(){
        StackDemo demo = new StackDemo(5);
        demo.push("A");
        demo.push("66");
        demo.push('c');
        demo.push(1L);
        demo.push(2.00);
//        demo.push(3.00);


        System.out.println(demo.flag);
        demo.pop();
        demo.pop();
        demo.pop();
        demo.pop();
        demo.show();
        System.out.println(demo.flag);
        System.out.println("///////");
        demo.push(2.00);
        demo.show();


    }

    class StackDemo{
        private int maxSize;
        private Object[] stack;
        private int flag = 0;

        public StackDemo(int maxSize) {
            this.maxSize = maxSize;
            stack = new Object[this.maxSize];
        }

        public Object pop(){
            if (empty()){
                System.out.println("the stack is empty.");
                return false;
            }
            return stack[--flag];
        }

        public void show(){
            for (int i = 0; i < flag; i++) {
                System.out.println(stack[i]);
            }
        }


        public boolean push(Object value){
            if (full()){
                System.out.println("the stack full");
                return false;
            }
            stack[flag++] = value;
            return true;
        }

        public boolean full(){
            return flag == maxSize;
        }

        public boolean empty(){
            return flag == 0;
        }





    }
}

File: kotlin-mybatis/src/test/java/pres/qianmuna/kt/Test1.kt
package pres.qianmuna.kt

import org.junit.Test
import pres.qianmuna.kt.config.MybatisFactoryConfig
import pres.qianmuna.kt.entity.BillModel
import pres.qianmuna.kt.entity.ProviderModel
import pres.qianmuna.kt.entity.UserModel
import pres.qianmuna.kt.mapper.BillMapping
import pres.qianmuna.kt.mapper.ProviderMapping
import pres.qianmuna.kt.mapper.UserMapping

/**
@author HJC
@date 2020/6/4  17:03
@description :
@version 1.0
谦谦君子 卑以自牧也
 */

class Test1 {


    /**
     * 查找
     */
    @Test
    fun m1(){

        //select user by id
        val sqlSession = MybatisFactoryConfig.sqlSession

        val mapper = sqlSession.getMapper(BillMapping::class.java)

        val mutableList = mapper.queryBillByNameAndProviderIdAndIsPay("%油%", 7, 2)

        println(mutableList)

        MybatisFactoryConfig.closeSqlSession

    }

    /**
     * 添加
     */
    @Test
    fun m2(){

        //select user by id
        val sqlSession = MybatisFactoryConfig.sqlSession

        val mapper = sqlSession.getMapper(ProviderMapping::class.java)

        val provier = ProviderModel(0,"1","1",
                "1","1","1","1","1","1")
        var i = mapper.addProvider(provier)

        //
        //提交啊
        sqlSession.commit()

        println(i)

        MybatisFactoryConfig.closeSqlSession

    }

    /**
     * 修改
     */
    @Test
    fun m3(){

        //select user by id
        val sqlSession = MybatisFactoryConfig.sqlSession

        val mapper = sqlSession.getMapper(ProviderMapping::class.java)

        val provier = ProviderModel(21,"8888","666",
                "7777","1","1","1","1","1")
        var i = mapper.updateProvider(provier)

        //
        //提交啊
        sqlSession.commit()

        println(i)

        MybatisFactoryConfig.closeSqlSession

    }

    /**
     * 删除
     */
    @Test
    fun m4(){

        //select user by id
        val sqlSession = MybatisFactoryConfig.sqlSession

        val mapper = sqlSession.getMapper(ProviderMapping::class.java)

        val i = mapper.deleteProviderById(21L)

        //提交啊
        sqlSession.commit()

        println(i)

        MybatisFactoryConfig.closeSqlSession

    }

    /**
     * 查找用户 地址
     */
    @Test
    fun m5(){

        //select user by id
        val sqlSession = MybatisFactoryConfig.sqlSession

        val mapper = sqlSession.getMapper(UserMapping::class.java)

        var list = mapper.queryUserAddress2()

        list?.forEach { println(it) }

        //提交啊


        MybatisFactoryConfig.closeSqlSession

    }
}

